gnome-initial-setup/pages/summary/gis-summary-page.c

```diff
@@ -339,6 +346,24 @@ gis_summary_page_constructed (GObject *object)
                             "lost forever when you restart or shutdown the computer.</b>"));
   }
 
+  /* Override the contents if needed */
+  summary_mode = gis_driver_conf_get_string (GIS_PAGE (page)->driver,
+                                             CONFIG_SUMMARY_GROUP,
+                                             CONFIG_SUMMARY_CONTENT_MODE_KEY);
+  if (g_strcmp0 (summary_mode, HACK_CONTENT_MODE) == 0) {
+    gtk_orientable_set_orientation (GTK_ORIENTABLE (priv->summary_contents_box),
+                                    GTK_ORIENTATION_HORIZONTAL);
+
+    gtk_image_set_from_resource (priv->image, "/org/gnome/initial-setup/ada.png");
```

Comment:
Personally I'd prefer to have this layout specified in the `.ui` file in a hidden-by-default widget and have the C code just toggle visibility depending on whether we're in hack mode or not. This is what I did for switching USB and DVD assets and I think that is neater. In theory it would mean you could preview/tweak the layout in Glade without having to rebuild and step through the FBE to test every change.

https://github.com/endlessm/gnome-initial-setup/blob/a3bdc8506e9f80eb7a6bb5f61edb353185caa695/gnome-initial-setup/pages/live-chooser/gis-live-chooser-page.ui#L76-L99

https://github.com/endlessm/gnome-initial-setup/blob/a3bdc8506e9f80eb7a6bb5f61edb353185caa695/gnome-initial-setup/pages/live-chooser/gis-live-chooser-page.c#L164-L167

Reply:
It's seems a bit overkill to me to do that when we have already an image + text showing. But I can still do it..

Reply:
I think this approach of changing the orientation and messing with the size requests is more error-prone than defining two sets of widgets. I could be wrong…